plz.el

```diff
@@ -961,10 +961,13 @@ Assumes that point is at beginning of HTTP response."
   "Return coding system for HTTP response in current buffer.
 HEADERS may optionally be an alist of parsed HTTP headers to
 refer to rather than the current buffer's un-parsed headers."
-  (let* ((headers (or headers (plz--headers)))
-         (content-type (alist-get 'content-type headers)))
-    (when content-type
-      (coding-system-from-name content-type))))
+  (when-let* ((headers (or headers (plz--headers)))
+              (raw-content-type (alist-get 'content-type headers))
+              (content-type (downcase raw-content-type))
```

Comment:
Alternatively, you could bind `case-fold-search` around the `string-match`; saves a string allocation.

Reply:
Yet more alternatively (for more brownie points?), did you consider using the `mail-parse` library?  See [`(info "(emacs-mime) Interface Functions")`](https://gnu.org/s/emacs/manual/html_node/emacs-mime/Interface-Functions.html).  For example:

```el
(require 'mail-parse)
(mail-content-type-get
 (mail-header-parse-content-type "text/javascript; charset=ISO-8859-1")
 'charset)
;; => "ISO-8859-1"
```

Reply:
> Alternatively, you could bind `case-fold-search` around the `string-match`; might save a string allocation.

Thanks, that's a good idea.  @josephmturner Would you like to add that, or should I?

Reply:
> Yet more alternatively (for more brownie points?), did you consider using the built-in `mail-parse` library?

No, I wasn't aware of that function.  Given the simplicity of this scenario, I think I'd prefer to not have to load that library.  Looking at the `rfc2231` library it is aliased to, while that library is certainly useful, the function `rfc2231-parse-string` which gets called looks large, complicated, and would probably be overkill for this.  I can't imagine it would help performance to use that function, either.  Thanks for pointing it out, though.  :)

Reply:
> I can't imagine it would help performance to use that function

Neither does making HTTP requests ;).

Reply:
Good idea @basil-conto !  I made this change, rebased, and force-pushed.